### PR TITLE
avoid the "duplicated promise" error which comes from the vue-router

### LINKED ISSUE #252: Error in log when clicking the current menu on the left menu
vue-router.esm.js?8c4f:2089 Uncaught (in promise) NavigationDuplicated {_name: "NavigationDuplicated", name: "NavigationDuplicated", message: "Navigating to current location ("/wallets/details") is not allowed", stack: "Error↵    at new NavigationDuplicated (webpack-int…node_modules/vue/dist/vue.runtime.esm.js:2178:14)"}

### PR DESCRIPTION
**Pull request by David0817**
( fix #252 )

### CODE REVIEW

**Comment by decentraliser** [comment]
isn't it a "NavigationDuplicated" error, as opposed to "duplicated promise"?
try adding a catch block to router.push instead
[reference](https://github.com/vuejs/vue-router/issues/2872)

image

**Comment by vincent-lee90** [comment]
@decentraliser 
`try...catch...` will not work, becase the root reason is that the promise API.
this will help you [https://github.com/vuejs/vue-router/issues/2881]

**Comment by decentraliser** [comment]
well, this issue is a regression, so I'm pretty sure it will work

please give it a shot

` @click="!currentAccount ? '' : $router.push({name: route.name}).catch(err => {})"`

**Comment by vincent-lee90** [comment]
before i thought that's how you let me is that :
```javascript
try{
     this.currentAccount && this.$router.push({ name: route.name })
}catch(e){
}
```
and now 
 `this.currentAccount && this.$router.push({ name: route.name }).catch(err => {})"` will work.

**Comment by vincent-lee90** [comment]
rebased pr177

File: src/components/PageNavigator/PageNavigatorTs.ts [hunk 1]
```diff
@@ -41,4 +42,8 @@ export class PageNavigatorTs extends Vue {
     await this.$store.dispatch('account/LOG_OUT')
     this.$router.push({name: 'accounts.login'})
   }
+  
+  public onPageNavigate(route: Route) {
+    this.currentAccount && this.$router.push({ name: route.name }).catch(()=>{/**/})
```
Comment (evias) [inline src/components/PageNavigator/PageNavigatorTs.ts:47]: This error should not be obfuscated. If it happens, it means that likely the **button** is not being disabled when the page is active.

Please, do not obfuscate the error, make sure it is _logged_, or that _it catches only this error._

An example exploit here would be to inject `hello-world` inside `route.name`. The router will throw an unknown route exception but your code will obfuscate it.

**Comment by vincent-lee90** [comment]
changed

**Pull request merged**